apps/charterafrica/src/lib/openAfrica.js

```diff
@@ -50,37 +73,83 @@ async function formatDatasets(data) {
     };
   });
 
+  const countries = Object.keys(groups || {}).sort((a, b) =>
+    a.localeCompare(b)
+  );
+  const tagsList = Object.keys(tags || {}).sort((a, b) => a.localeCompare(b));
+
   return {
-    datasets: formattedDatasets,
-    countries: Object.keys(allCountries).map((country) => country),
-    tags: Object.keys(allTags).map((tag) => tag),
     count,
+    datasets,
+    documents,
+    countries,
+    tags: tagsList,
+    totalPages: Math.ceil(count / PageSize),
   };
 }
 
 export default async function fetchDatasets(organization, query = {}) {
-  const { tags = [], ...other } = query;
-  const tagsQuery = tags.length ? `tags:(${tags.join(" OR ")})` : "";
+  const { tags = [], countries = [], page = 1, ...other } = query;
+  const tagsQuery = tags.length
+    ? `tags:(${tags.map((tag) => `"${tag}"`).join(" OR ")})`
+    : null;
+  const countriesQuery = countries.length
+    ? `groups:(${countries.map((country) => `"${country}"`).join(" OR ")})`
```

Comment:
Instead of converting to array of string first only to call `join`, I think you should be able to use `.reduce` directly on `tags` or `countries` to convert them to a single string.